### PR TITLE
Experimental Parser: Change Python for Rust

### PR DESCRIPTION
**Pull request by nathaniel-may**
Merges [PR#3497](https://github.com/fishtown-analytics/dbt/pull/3497)

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [x] I have run this code in development and it appears to resolve the stated issue
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

File: core/setup.py [hunk 1]
```diff
@@ -73,10 +73,9 @@ def read(fname):
         'networkx>=2.3,<3',
         'packaging~=20.9',
         'sqlparse>=0.2.3,<0.4',
-        'tree-sitter==0.19.0',
-        'tree-sitter-jinja2==0.1.0a1',
-        'typing-extensions>=3.7.4,<3.8',
-        'werkzeug>=0.15,<2.0',
+        'dbt-extractor==0.2.0',
+        'typing-extensions>=3.7.4,<3.11',
+        'werkzeug>=0.15,<3.0',
```
Comment (kwigley) [inline core/setup.py:78]: hmm, I'm not opposed, _but_ should we widen this dependency range in this PR? I think we did on `develop`, but not in `0.20.latest`

**Review by kwigley — APPROVED** [review]

File: core/setup.py (hunk 1 above)
Comment (nathaniel-may) [reply core/setup.py:78]: It looks like it matches develop. Am I missing something?

Comment (nathaniel-may) [reply core/setup.py:78]: Oh you mean that we should consider putting it back to the previous narrower range here.

Comment (kwigley) [reply core/setup.py:78]: ya ya, `'werkzeug>=0.15,<2.0'`

Comment (nathaniel-may) [reply core/setup.py:78]: I'm not familiar with our usage of that package. What are the pros and cons in your mind?

Comment (kwigley) [reply core/setup.py:78]: It is a dependency of the rpc server.. I'm not sure we've thoroughly tested any versions of `werkzeug`> `2.0.0` though

Comment (nathaniel-may) [reply core/setup.py:78]: If we don't need it I'm happy to reduce it. Since I merged this, do you want to open a new PR with that?

**Pull request merged**